applications/fishing-map/src/features/map/map.slice.ts

```diff
@@ -29,12 +34,50 @@ type Fetch4WingInteractionThunk = {
 }
 export const fetch4WingInteractionThunk = createAsyncThunk(
   'map/fetchInteraction',
-  async ({ dataset, datasetConfig }: Fetch4WingInteractionThunk) => {
+  async (
+    { dataset, datasetConfig }: Fetch4WingInteractionThunk,
+    { getState, signal, rejectWithValue }
+  ) => {
     const url = resolveEndpoint(dataset, datasetConfig)
     const datasetId = datasetConfig.query?.find((query) => query.id === 'datasets')?.value as any
+    let vesselsForDataset: ExtendedFeatureVessel[]
     if (url) {
-      const vesselsByDataset = await GFWAPI.fetch<Record<string, ExtendedFeatureVessel[]>>(url)
-      const vesselsForDataset = vesselsByDataset[datasetId]
+      const vesselsByDataset = await GFWAPI.fetch<Record<string, ExtendedFeatureVessel[]>>(url, {
+        signal,
+      })
+      vesselsForDataset = vesselsByDataset[datasetId]
+      const infoDatasetId = getRelatedDatasetByType(dataset, VESSELS_DATASET_TYPE)?.id
+      if (infoDatasetId) {
+        const infoDataset = selectDatasetById(infoDatasetId)(getState() as RootState)
+        if (infoDataset) {
+          const infoDatasetConfig = {
+            endpoint: 'carriers-multiple-vessel',
+            datasetId: infoDataset.id,
+            params: [],
+            query: [
+              { id: 'datasets', value: infoDataset.id },
+              {
+                id: 'ids',
+                value: vesselsForDataset.slice(0, MAX_TOOLTIP_VESSELS).map((v) => v.id),
+              },
+            ],
+          }
+          const infoUrl = resolveEndpoint(infoDataset, infoDatasetConfig)
+          if (infoUrl) {
+            // TODO create search API results response
+            const vesselsInfo = await GFWAPI.fetch<any>(infoUrl, { signal })
```

Comment:
What's `signal` for?

Reply:
it is an [AbortController](https://developer.mozilla.org/en-US/docs/Web/API/AbortController/signal) instance property that can be used to abort a fetch request